port/cpl_aws.cpp

```diff
@@ -1746,6 +1758,172 @@ bool VSIS3HandleHelper::GetOrRefreshTemporaryCredentialsForSSO(
     return false;
 }
 
+/************************************************************************/
+/*                  GetCredentialsFromProcess()                         */
+/************************************************************************/
+
+static bool GetCredentialsFromProcess(const std::string &osCredentialProcess,
+                                      std::string &osSecretAccessKey,
+                                      std::string &osAccessKeyId,
+                                      std::string &osSessionToken)
+{
+    CPLDebug(AWS_DEBUG_KEY, "Executing credential_process: %s",
+             osCredentialProcess.c_str());
+
+    CPLStringList aosOptions;
+    aosOptions.SetNameValue("TIMEOUT", "30");
+
+    FILE *fp = popen(osCredentialProcess.c_str(), "r");
+    if (fp == nullptr)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined,
+                 "Failed to execute credential_process: %s",
+                 osCredentialProcess.c_str());
```

Comment:
Should we include `VSIStrerror(errno)` in the error message to show the reason why popen failed?